### PR TITLE
S3 and KMS composition examples

### PR DESCRIPTION
**Pull request by vara-bonthu**
### What does this PR do?

:white_check_mark: S3 bucket composition with KMS Key with **Terrajet provider**

:white_check_mark: Managed resources examples for **Terrajet provider**
     - S3 Bucket
     - KMS Key

:white_check_mark: Kubernetes provider for Terraform bootstrap scripts. Fix for #15 

:white_check_mark: Kubernetes provider examples with docs

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)

- [x] Yes, I have added a new example under [examples](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/examples) to support my PR

- [x] Yes, I have updated the [docs](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/doc) for this feature

- [x] Yes, I have linked to an issue or feature request (applicable to PRs that solves a bug or a feature request)

**Note**:
 - Not all the PRs require examples and docs
 - We prefer small, well tested pull requests. Please ensure your pull requests are self-contained, and commits are squashed

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: compositions/terrajet-aws-provider/s3/s3-composition.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: s3bucket.jet.awsblueprints.io
+  labels:
+    awsblueprints.io/provider: jet-aws
+    awsblueprints.io/environment: dev
+    s3.awsblueprints.io/configuration: standard
+    crossplane.io/xrd: xs3buckets.jet.awsblueprints.io
```
Comment (nabuskey) [inline compositions/terrajet-aws-provider/s3/s3-composition.yaml:12]: Do we care about this?

File: examples/terrajet-aws-provider/managed-resources/kms.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (nabuskey) [inline examples/terrajet-aws-provider/managed-resources/kms.yaml:1]: What do you think about having examples for managed resources? Crossplane's use cases pretty much expects compositions to be used. I feel like managed resources examples can be found in compositions.

**Review by nabuskey — APPROVED** [review]

File: compositions/terrajet-aws-provider/s3/s3-composition.yaml (hunk 1 above)
Comment (vara-bonthu) [reply compositions/terrajet-aws-provider/s3/s3-composition.yaml:12]: Not really but I kept it for consistency with other compositions

File: compositions/terrajet-aws-provider/s3/s3-composition.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+# SPDX-License-Identifier: MIT-0
+
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: s3bucket.jet.awsblueprints.io
+  labels:
+    awsblueprints.io/provider: jet-aws
+    awsblueprints.io/environment: dev
+    s3.awsblueprints.io/configuration: standard
```
[the remark this answers is not in the document]
Comment (vara-bonthu) [reply compositions/terrajet-aws-provider/s3/s3-composition.yaml:11]: Agree. Will update

**Pull request merged**